src/core/thread/link_quality.cpp

```diff
@@ -140,6 +140,24 @@ RssAverager::InfoString RssAverager::ToString(void) const
     return string;
 }
 
+void LqiAverager::Reset(void)
+{
+    mCount   = 0;
+    mAverage = 0;
+}
+
+void LqiAverager::Add(uint8_t aLqi)
+{
+    uint16_t oldAverage = mAverage;
+    uint8_t  count      = OT_MIN((1 << kCoeffBitShift), mCount + 1);
```

Comment:
note that here `mCount` can be 255 and in `count` can overflow back to 0. 
Maybe better to assign `count` to `mCount` after `mCount` is increased?